src/engine/plugins/tile.cpp

```diff
@@ -181,20 +287,318 @@ Status TilePlugin::HandleRequest(const api::TileParameters &parameters, std::str
+        for (const auto &source_ebn : edge_based_node_info)
+        {
+            // Grab a copy of the geometry leading up to the intersection.
+            std::vector<NodeID> first_geometry;
+            facade.GetUncompressedGeometry(source_ebn.second.packed_geometry_id, first_geometry);
+
+            // We earlier saved the source and target intersection nodes for every road section.
+            // We can use the target node to find all road sections that lead away from
+            // the intersection, and thus
+            // in the graph after our main
+            for (const auto &target_ebn : outgoing_edges[source_ebn.second.target_intersection])
+            {
+                // Ignore u-turns for now
+                if (edge_based_node_info.at(target_ebn).target_intersection ==
+                    source_ebn.second.source_intersection)
+                    continue;
+
+                // Find the connection between our source road and the target node
+                EdgeID smaller_edge_id = facade.FindSmallestEdge(
+                    source_ebn.first, target_ebn, [](const contractor::QueryEdge::EdgeData &data) {
+                        return data.forward;
+                    });
+
+                // Depending on how the graph is constructed, we might have to look for
+                // a backwards edge instead.  They're equivalent, just one is available for
+                // a forward routing search, and one is used for the backwards dijkstra
+                // steps.  Their weight should be the same, we can use either one.
+                // If we didn't find a forward edge, try for a backward one
+                if (SPECIAL_EDGEID == smaller_edge_id)
+                {
+                    smaller_edge_id = facade.FindSmallestEdge(
+                        target_ebn, source_ebn.first, [](const contractor::QueryEdge::EdgeData &data) {
+                            return data.backward;
+                        });
+                }
+
+                // If no edge was found, it means that there's no connection between these nodes,
+                // due to oneways or turn restrictions. Given the edge-based-nodes that
+                // we're examining here, we *should* only find directly-connected edges, not
+                // shortcuts
+                if (smaller_edge_id != SPECIAL_EDGEID)
+                {
+                    // Check to see if it was a shortcut edge we found.  This can happen
+                    // when exactly?  Anyway, unpack it and get the first "real" edgedata
+                    // out of it, which should represent the first hop, which is the one
+                    // we want to find the turn.
+                    auto data = facade.GetEdgeData(smaller_edge_id);
+                    if (data.shortcut)
+                    {
+                        std::vector<contractor::QueryEdge::EdgeData> unpacked_shortcut;
```

Comment:
Hmm thinking of this, there is a trick we can do without re-writing the UnpackEdges function: use a boost function output iterator with a lambda instead of writing to a vector and then using only the first one (`boost::make_function_output_iterator`).